test/integration/testExecute.js

```diff
@@ -396,3 +396,224 @@ describe('Execute test - variant', function () {
     it(testCase.name, createItCallback(testCase, rowAsserts));
   });
 });
+
+describe('Execute test with Pool', function ()
+{
+  var connectionPool;
+  var createNodeTSQL = 'create or replace table NodeT(colA number, colB varchar)';
+  var selectAllSQL = 'select * from NodeT';
+  var insertNodeTSQL = 'insert into NodeT values(1, \'a\')';
+  var updateNodeTSQL = 'update NodeT set COLA = 2, COLB = \'b\' where COLA = 1';
+  var dropNodeTSQL = 'drop table if exists NodeT';
+
+  before(function (done)
+  {
+    connectionPool = testUtil.createConnectionPool();
+    done();
+  });
+
+  it('testSimpleInsert', function (done)
+  {
+    async.series(
+      [
+        function (callback)
+        {
+          testUtil.executeCmdUsePool(connectionPool, createNodeTSQL, callback);
+        },
+        function (callback)
```

Comment:
we could, but I noticed that when running with Nodejs17 and Nodejs18 on github, the asycc/await execute is causing some error.
https://github.com/snowflakedb/snowflake-connector-nodejs/actions/runs/5490079306/jobs/10005081663?pr=564
I haven't figure out why because it succeeded when I test with my local machine using nodejs17 and nodejs18. So I cannot reproduce.
I would suggest to stay with what it is right now. But if you think it is better to change to async/await, let me know.